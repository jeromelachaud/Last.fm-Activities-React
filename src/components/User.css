/* Usercard */
.User {
  display: inline-block;
  margin: 5px auto;
  font-size: 2rem;
}

.User__link {
  display: inherit;;
  color: #fff;
  background-color: rgba(255,255,255,0.4);
}

.User__info {
  margin: 0 auto;
  padding: 1rem 0.7rem 0.7rem 0;
}

.User__info li span{
  font-size: 1.5rem;
  font-style: italic;
  margin-right: 5px;
}
