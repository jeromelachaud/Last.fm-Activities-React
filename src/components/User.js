import React from 'react';
import styled from 'styled-components';
import moment from 'moment';
import {
  fetchUser
} from '../actions/action-creators';

const UserWrapper = styled.div`
  display: inline-block;
  margin: 5px auto;
  font-size: 2rem;
`;

const UserLink = styled.a`
  display: inherit;;
  color: #fff;
  background-color: rgba(255,255,255,0.4);
`;

const UserInfo = styled.ul`
  margin: 0 auto;
  padding: 1rem 0.7rem 0.7rem 0;

  li {
    font-size: 1.5rem;
    font-style: italic;
    margin-right: 5px;
  }
`;


let User = React.createClass ({

  propTypes: {
    user: React.PropTypes.object.isRequired
  },

  componentWillMount() {
    fetchUser();
  },

  render() {
    const {
      user
    } = this.props;

    const {
      image,
      name,
      playcount,
      registered,
      url
    } = user;

    const registeredElement = moment.unix(registered['#text']).format('MM/DD/YYYY');

    return (
      <UserWrapper>
        <UserLink
          href={url} target="_blank">
          <div>
            <img
              src={image['3']['#text']}
              alt={name} />
          </div>
          <UserInfo>
            <li>{name}</li>
            <li>{playcount}</li>
            <li>{registeredElement}</li>
          </UserInfo>
        </UserLink>
      </UserWrapper>
    );
  }
});

export default User;
