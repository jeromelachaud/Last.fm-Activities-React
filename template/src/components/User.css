.Name {
  color: white;
  font-size: 2rem;
}

.Logo {
  width: 5rem;
}

/* Usercard */
.User {
  font-size: 2rem;
  width: 25rem;
  margin: 5px auto;
  color: #fff;
}

@media screen and (min-width: 860px) {
  .User {
    width: auto;
    border: 0;
  }
}
