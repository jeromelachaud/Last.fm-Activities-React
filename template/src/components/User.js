import React from 'react';
import './User.css';
import moment from 'moment';

let User = React.createClass ({

  render() {
    const {
      user
    } = this.props;

    const {
      name,
      playcount,
      registered,
      image,
      url
    } = user;

    const registeredElement = moment.unix(registered['#text']).format('MM/DD/YYYY');

    return (
      <div
        className="User">
        <a
          className="User__link"
          href={url} target="blanck">
          <div>
            <img src={image[3]['#text']} />
          </div>
          <ul
            className="User__info">
            <li>{name}</li>
            <li>{playcount}</li>
            <li>{registeredElement}</li>
          </ul>
        </a>
      </div>
    );
  }
});

User.propTypes = {
  user: React.PropTypes.shape({})
};

export default User;
