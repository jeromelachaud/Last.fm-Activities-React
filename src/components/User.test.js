import React from 'React';
import {
  shallow
} from 'enzyme';
import toJson from 'enzyme-to-json';
import User from './User';

const user = {
  'user':'a user'
};

describe('<User /> componemts', () => {
  const wrapper = shallow(<User user={user}/>);

  it('should render as expected', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
