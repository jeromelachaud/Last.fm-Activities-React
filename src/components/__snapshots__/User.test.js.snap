exports[`<User /> componemts should render as expected 1`] = `
<styled.div>
  <styled.a
    href=""
    target="_blank">
    <div>
      <img
        alt=""
        src="" />
    </div>
    <styled.ul>
      <li />
      <li />
      <li>
        01/01/1970
      </li>
    </styled.ul>
  </styled.a>
</styled.div>
`;
